Tests/ProcessLib/TestParameter.cpp

```diff
@@ -0,0 +1,176 @@
+{
+    const char xml[] =
+            "<parameter>"
+            "<type>PropertyIndex</type>"
+            "<property_name>MaterialIDs</property_name>"
+            "<index_values><index>0</index><value>0</value></index_values>"
+            "<index_values><index>1</index><value>100</value></index_values>"
+            "<index_values><index>3</index><value>300</value></index_values>"
+            "</parameter>";
+    auto const ptree = readXml(xml);
+
+    std::unique_ptr<MeshLib::Mesh> mesh(MeshLib::MeshGenerator::generateLineMesh(4u, 1.0));
+    std::vector<int> mat_ids({0,1,2,3});
+    MeshLib::addPropertyToMesh(*mesh, "MaterialIDs", MeshLib::MeshItemType::Cell, 1, mat_ids);
+
+    Callbacks cbs;
+    {
+        auto const conf = makeConfigTree(ptree, cbs);
+        std::unique_ptr<ProcessLib::ParameterBase> parameter_base = ProcessLib::createPropertyIndexParameter(conf.getConfigSubtree("parameter"), *mesh);
+        EXPECT_ERR_WARN(cbs, false, false);
+
+        auto parameter = dynamic_cast<ProcessLib::Parameter<double>*>(parameter_base.get());
+        double t = 0;
+        ProcessLib::SpatialPosition x;
+        x.setElementID(0);
+        ASSERT_EQ(0.0, (*parameter)(t, x)[0]);
+        x.setElementID(1);
+        ASSERT_EQ(100.0, (*parameter)(t, x)[0]);
+        x.setElementID(2);
+        ASSERT_TRUE((*parameter)(t, x).empty());
+        x.setElementID(3);
+        ASSERT_EQ(300.0, (*parameter)(t, x)[0]);
+
+
+    } // ConfigTree destroyed here
+    EXPECT_ERR_WARN(cbs, false, false);
+}
+
+TEST(ProcessLib_Parameter, PropertyIndexParameterNode)
+{
+    const char xml[] =
+            "<parameter>"
+            "<type>PropertyIndex</type>"
+            "<property_name>PointGroupIDs</property_name>"
+            "<index_values><index>0</index><value>0</value></index_values>"
+            "<index_values><index>1</index><value>100</value></index_values>"
+            "<index_values><index>3</index><value>300</value></index_values>"
+            "</parameter>";
+    auto const ptree = readXml(xml);
+
+    std::unique_ptr<MeshLib::Mesh> mesh(MeshLib::MeshGenerator::generateLineMesh(4u, 1.0));
+    std::vector<int> group_ids({0,1,2,3,4});
+    MeshLib::addPropertyToMesh(*mesh, "PointGroupIDs", MeshLib::MeshItemType::Node, 1, group_ids);
+
+    Callbacks cbs;
+    {
+        auto const conf = makeConfigTree(ptree, cbs);
+        std::unique_ptr<ProcessLib::ParameterBase> parameter_base = ProcessLib::createPropertyIndexParameter(conf.getConfigSubtree("parameter"), *mesh);
+        EXPECT_ERR_WARN(cbs, false, false);
+
+        auto parameter = dynamic_cast<ProcessLib::Parameter<double>*>(parameter_base.get());
+        double t = 0;
+        ProcessLib::SpatialPosition x;
+        x.setNodeID(0);
+        ASSERT_EQ(0.0, (*parameter)(t, x)[0]);
+        x.setNodeID(1);
+        ASSERT_EQ(100.0, (*parameter)(t, x)[0]);
+        x.setNodeID(2);
+        ASSERT_TRUE((*parameter)(t, x).empty());
```

Comment:
Wouldn't it be better to `OGS_FATAL` with a proper error message if the material ID is not found?